protoc-gen-connect-kotlin/src/main/kotlin/com/connectrpc/protocgen/connect/Generator.kt

```diff
@@ -524,3 +539,36 @@ private fun String.packageToDirectory(): String {
     }
     return dir
 }
+
+private fun TypeSpec.Builder.addServiceDeprecation(
+    service: Descriptors.ServiceDescriptor,
+    file: Descriptors.FileDescriptor
+): TypeSpec.Builder {
+    if (service.options.deprecated) {
+        this.addAnnotation(
+            AnnotationSpec.builder(Deprecated::class)
+                .addMember("%S", "The underlying service is marked deprecated.")
+                .build()
+        )
+    } else if (file.options.deprecated) {
+        this.addAnnotation(
+            AnnotationSpec.builder(Deprecated::class)
+                .addMember("%S", "The underlying file is marked deprecated.")
```

Comment:
Ditto. This is the main one that I think might be confusing since it could be very unclear to a reader what "underlying file" means.
```suggestion
                .addMember("%S", "The Protobuf source file that defines this service is deprecated.")
```